### PR TITLE
PaperLib Async TP

### PR DESCRIPTION
**Pull request by JT122406**
Moves all Teleporting to PaperLib async teleport to increase performance for servers running plugin

### CODE REVIEW

**Comment by Tigerpanzer02** [comment]
I would prefer creating a new method on VersionUtils called teleport(entity, location) due to 3 reasons
1. If something on the lib changes or doesn't work in the future we can change it way faster 
2. We could handle misleading teleports with a message (documentation paperlib)
3. Our plugins which are based on the core use a method from the corr instead of another library, easier to maintain

To discuss if VersionUtils is the right place for it as it basically detects changed on different Minecraft version but as paper is also included in it with isPaper it would be totally fine so far I see to include it into VersionUtils

**Comment by JT122406** [comment]
> I would prefer creating a new method on VersionUtils called teleport(entity, location) due to 3 reasons
> 
> 1. If something on the lib changes or doesn't work in the future we can change it way faster 
> 
> 2. We could handle misleading teleports with a message (documentation paperlib)
> 
> 3. Our plugins which are based on the core use a method from the corr instead of another library, easier to maintain
> 
> 
> 
> To discuss if VersionUtils is the right place for it as it basically detects changed on different Minecraft version but as paper is also included in it with isPaper it would be totally fine so far I see to include it into VersionUtils

Will implement this in the coming days when I have time

File: MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/utils/version/VersionUtils.java [hunk 1]
```diff
@@ -63,7 +59,7 @@
 @SuppressWarnings("deprecation")
 public final class VersionUtils {
 
-  private static boolean isPaper = false, isParticleBuilderSupported = false;
+  private static boolean isPaper, isParticleBuilderSupported = false;
```
Comment (Tigerpanzer02) [inline MiniGamesBox Classic/src/main/java/plugily/projects/minigamesbox/classic/utils/version/VersionUtils.java:62]: so particlebuilder could be also the false removed, if starting with one so do for all

File: pom.xml [hunk 2]
```diff
@@ -77,6 +77,13 @@
             <version>1.17.1-R0.1-SNAPSHOT</version>
             <scope>provided</scope>
         </dependency>
+        <!-- Paper Lib-->
+        <dependency>
+            <groupId>io.papermc</groupId>
+            <artifactId>paperlib</artifactId>
+            <version>1.0.7</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (Tigerpanzer02) [inline pom.xml:86]: PaperLib is not already included in paper server software, right?

**Review by Tigerpanzer02 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (JT122406) [reply pom.xml:86]: It is included in the Server Software but not in the paper dependency

Comment (JT122406) [reply pom.xml:86]: Hence we have to add it

**Comment by JT122406** [comment]
I made the changes

File: pom.xml (hunk 2 above)
Comment (Tigerpanzer02) [reply pom.xml:86]: Yeah but if it's on the software we would not need to compile it

Comment (Tigerpanzer02) [reply pom.xml:86]: Can you validate that @JT122406

Comment (JT122406) [reply pom.xml:86]: um yeah I can check on it

Comment (JT122406) [reply pom.xml:86]: only needs to be provided I changed it

**Comment by JT122406** [comment]
@Tigerpanzer02 Should be fixed now

File: pom.xml (hunk 2 above)
Comment (Tigerpanzer02) [reply pom.xml:86]: Awesome

**Review by Tigerpanzer02 — CHANGES_REQUESTED** [review]
Shading patter not needed if provided

**Comment by JT122406** [comment]
Lol my bad forgot I shaded it @Tigerpanzer02

**Comment by Tigerpanzer02** [comment]
Seems fine, mergin it ;) Thanks for your contribution

**Pull request merged**